src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryDeployer.java

```diff
@@ -175,6 +176,21 @@ Path getApplication(AppDeploymentRequest request) {
 			.doOnError(e -> logger.error(String.format("Retry operation on getStatus failed for %s.  Max retry time %sms", id, statusTimeout)));
 	}
 
+	protected void deleteLocalApplicationResourceFile(AppDeploymentRequest appDeploymentRequest) {
+		try {
+			if (!appDeploymentRequest.getResource().getURI().toString().startsWith("docker:")) {
```

Comment:
This is going to create other side effects that we have observed, for example, in our FTP ingest sample, we should not have to download over and over again the same jar file for the task.   That is not just inefficient, but also exposed an issue were we should be adding retry advice to the download process because a 'thundering herds' problem resulted in unsuccessful downloads.  For maven based resources, I believe we should leave the file as is and not delete it.  If user are in development and operating via `SNAPSHOT` then the disk space is largely conserved. If users are using standard versioning, e.g. 1.0.1, 1.0.2, 1.0.3 and so on, then it can become more likely that disk space will be exhausted, in which case users should really be using volume services on PCF.